x-pack/plugins/observability/public/pages/alert_details/components/alert_details.tsx

```diff
@@ -27,19 +31,31 @@ import { paths } from '../../../config/paths';
 
 export function AlertDetails() {
   const {
+    uiSettings,
     http,
     cases: {
       helpers: { canUseCases },
       ui: { getCasesContext },
     },
+    triggersActionsUi: { ruleTypeRegistry },
   } = useKibana<ObservabilityAppServices>().services;
+
   const { ObservabilityPageTemplate, config } = usePluginContext();
   const { alertId } = useParams<AlertDetailsPathParams>();
   const [isLoading, alert] = useFetchAlertDetail(alertId);
-
+  const [ruleTypeModel, setRuleTypeModel] = useState<RuleTypeModel | null>(null);
   const CasesContext = getCasesContext();
   const userCasesPermissions = canUseCases();
+  const { rule } = useFetchRule({
+    ruleId: alert?.fields[ALERT_RULE_UUID],
+    http,
+  });
 
+  useEffect(() => {
+    if (alert) {
+      setRuleTypeModel(ruleTypeRegistry.get(alert?.fields[ALERT_RULE_TYPE_ID]!));
```

Comment:
You might not need `useEffect` and `useState` in this instance.

```
const ruleTypeModel = alert ? ruleTypeRegistry.get(alert?.fields[ALERT_RULE_TYPE_ID]) : undefined
```

Would also work.

Reply:
Also `ruleTypeRegistry` [will throw an error if the passed rule type can't be found](https://github.com/elastic/kibana/blob/main/x-pack/plugins/triggers_actions_ui/public/application/type_registry.ts:L47), so if you want to be safe it might be better to do something like:

```
const getRuleTypeModel = (ruleTypeId: string | undefined): RuleTypeModel<RuleTypeParams> | undefined => {
    try {
      return ruleTypeRegistry.get(alert?.fields[ALERT_RULE_TYPE_ID]);
    } catch (error) {
      return undefined
    }
}

const ruleTypeModel = getRuleTypeModel(alert?.fields[ALERT_RULE_TYPE_ID]);
```

Reply:
@CoenWarmer will check that, and see if this is the behavior that we would like to have.
Regarding the suggestion for the `getRuleTypeModel ` function, it doesn't have access to the Alert object as it's outside the component. I'm not sure about exposing all these arguments.